bayesian/api/api_v2.py

```diff
@@ -172,6 +180,113 @@ def get(ecosystem, package, version):
+
+        # 7. Default case of HTTP error
+        logger.error('Error :: {} => {}'.format(response_status, response_data))
+        raise HTTPError(response_status, error=response_data)
+
+    @staticmethod
+    def post():
+        """Handle /api/v2/stack-analyses POST REST API."""
+        # 1. Read mandatory params
+        manifest = request.files.get('manifest') or None
+        file_path = request.form.get('file_path') or None
+        ecosystem = request.form.get('ecosystem') or None
+
+        logger.debug('Mandatory params :: manifest: {} file_path: {} ecosystem: {}'.format(
+            manifest, file_path, ecosystem))
+
+        # 2. Read optional params and set default value as per V2 spec.
+        show_transitive = request.form.get('show_transitive') or 'true'
+
+        logger.info('Optional params :: show_transitive: {}'.format(show_transitive))
+
+        # 3. Validate post params.
+        # Manifest is mandatory and must be a string.
+        if manifest is None or not resolved_files_exist(manifest.filename):
+            error_message = 'Error processing request. Manifest is missing its value {} is ' \
+                            'invalid / not supported'.format(manifest)
+            logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # File path is mandatory and must be a string.
+        if file_path is None or not isinstance(file_path, str):
+            error_message = 'Error processing request. File path is missing / its value ' \
+                            '{} is invalid'.format(file_path)
+            logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Ecosystem  is mandatory and must be a string.
+        if ecosystem is None or not isinstance(ecosystem, str):
+            error_message = 'Error processing request. Ecosystem is missing / its value {} ' \
+                            'is invalid'.format(ecosystem)
+            logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Ecosystem should be a valid value
+        if not check_for_accepted_ecosystem(ecosystem):
+            error_message = 'Error processing request. {} ecosystem is not supported'.format(
+                ecosystem)
+            logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # 4. Build manifest info from manifest file and path. It read content in utf-8 encoding.
+        manifest_file_info = {
+            'filename': manifest.filename,
+            'filepath': file_path,
+            'content': manifest.read().decode('utf-8')
+        }
+        logger.info(manifest_file_info)
+
+        # 5. Initiate stack analyses object
+        sa = StackAnalyses(None, ecosystem, manifest_file_info, show_transitive)
+
+        # 6. Post request
+        response_status, response_data = sa.post_request()
+        if response_status == 200 or response_status == 202:
```

Comment:
Do we have 202 use case for POST?

Reply:
No, we don't this, I will correct.

Reply:
This is fixed in commit 'Review comment to HTTPError from base classes is fixed.'